src/stores/index.js

```diff
@@ -851,23 +896,38 @@ const Store = Reflux.createStore({
     // Set the connection's app name to the electron app name of Compass.
     connection.appname = electron.remote.app.getName();
 
-    const dataService = new DataService(connection);
+    if (!this.state.isConnecting) {
```

Comment:
i think is probably easier at this point if we use the connection attempt as source of truth. `!this.currentConnectionAttempt` and make sure it's always reset to null if the connection is not in progress.